ddtrace/internal/datadog/profiling/dd_wrapper/src/sample_manager.cpp

```diff
@@ -80,6 +80,8 @@ Datadog::SampleManager::postfork_child()
         // and the fork happened in the middle of that operation.
         sample_pool = std::make_unique<SynchronizedSamplePool>(sample_pool_capacity);
     }
+
+    Datadog::Sample::profile_state.one_time_init_impl(type_mask, max_nframes);
```

Comment:
In child process, calling `one_time_init_impl()` again from here results in allocating  `libdd Profile` objects. If the parent had succeeded calling this, then the child doesn't free the existing one and overwrites the variable, `cur_profile` and `last_profile`, line 178 and 182. Resulting in leaking the inner profile. 

Line 178 and 182 from one_time_init_impl

https://github.com/DataDog/dd-trace-py/pull/12865/files#diff-1af87bc2fc4ba5a4e8f4b172318d261b0d53039758557a2a4730c4def9ee42caR178-R182

Reply:
Good catch, taking a look

Reply:
Looked at this--I don't think one_time_init_impl() actually creates those objects.  `cur_profile` et al are allocated at static initialization time.

*however* those objects should be reset, and your commend made me realize that the postfork handler for this class isn't being called.  I reorganized things so the proper operations are called now from the sample manager. WDYT?